### PR TITLE
Added Atmosphere Python interface

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

This PR is part of this mete ticket https://github.com/ignitionrobotics/sdformat/issues/931

## Summary

Added Atmosphere Python interface

## Test it

```python
from sdformat import Atmosphere

Atmosphere = Atmosphere()
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/sdf/Atmosphere.hh [hunk 1]
```diff
@@ -95,6 +95,12 @@ namespace sdf
     /// \return True if this instance equals the given atmosphere.
     public: bool operator==(const Atmosphere &_atmosphere);
 
+    /// \brief Equality operator that returns true if this atmosphere
+    /// instance equals the given atmosphere instance.
+    /// \param[in] _atmosphere Atmosphere instance to compare.
+    /// \return True if this instance equals the given atmosphere.
+    public: bool operator==(const Atmosphere &_atmosphere) const;
```
Comment (azeey) [inline include/sdf/Atmosphere.hh:102]: I think it should be safe to remove the old function.

File: python/src/sdf/pyAtmosphere.cc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    .def("temperature_gradient", &sdf::Atmosphere::TemperatureGradient,
+         "Get the temperature gradient with respect to increasing "
+         "altitude in units of K/m.")
+    .def("set_temperature_gradient", &sdf::Atmosphere::SetTemperatureGradient,
+         "Set the temperature gradient with respect to increasing "
+         "altitude in units of K/m.")
+    .def("pressure", &sdf::Atmosphere::Pressure,
+         "Get the pressure at sea level in pascals.")
+    .def("set_pressure", &sdf::Atmosphere::SetPressure,
+         "et the pressure at sea level in pascals.")
```
Comment (azeey) [inline python/src/sdf/pyAtmosphere.cc:59]: nit
```suggestion
         "Set the pressure at sea level in pascals.")
```

File: python/src/sdf/pyAtmosphere.cc [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    .def("__copy__", [](const sdf::Atmosphere &self) {
+      return sdf::Atmosphere(self);
+    })
+    .def("__deepcopy__", [](const sdf::Atmosphere &self, pybind11::dict) {
+      return sdf::Atmosphere(self);
+    }, "memo"_a);
+
+    pybind11::enum_<sdf::AtmosphereType>(geometryModule, "AtmosphereType")
```
Comment (azeey) [inline python/src/sdf/pyAtmosphere.cc:67]: Should this be in the `sdformat` namespace?

Comment (ahcorde) [reply python/src/sdf/pyAtmosphere.cc:67]: you should access this enum:

```python
Atmosphere.AtmosphereType.ADIABATIC
```

Comment (azeey) [reply python/src/sdf/pyAtmosphere.cc]: Right, but that is different from the C++ equivalent `sdf::AtmosphereType` where it's directly under `sdf`, not `sdf::Atmosphere::AtmosphereType`. I'm just asking if we should try to closely match the C++ API or not.

Comment (ahcorde) [reply python/src/sdf/pyAtmosphere.cc]: for now, I'm using this way with all the python bindings, we can make a decision at the end and change all of them at once.

**Review by azeey — APPROVED** [review]

**Pull request merged**